Frameworks/Foundation/NSPointerArray.mm

```diff
@@ -14,146 +14,380 @@
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (self = [self init]) {
+        _pointerFunctions = [functions copy];
+    }
+    return self;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (NSPointerArray*)pointerArrayWithOptions:(NSPointerFunctionsOptions)options {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSPointerArray alloc] initWithOptions:options] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (NSPointerArray*)pointerArrayWithPointerFunctions:(NSPointerFunctions*)functions {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSPointerArray alloc] initWithPointerFunctions:functions] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (NSPointerArray*)strongObjectsPointerArray {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self pointerArrayWithOptions:NSPointerFunctionsStrongMemory | NSPointerFunctionsObjectPointerPersonality];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (NSPointerArray*)weakObjectsPointerArray {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self pointerArrayWithOptions:NSPointerFunctionsWeakMemory | NSPointerFunctionsObjectPointerPersonality];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+*/
+- (NSUInteger)count {
+    return _count;
+}
+
+/**
+ @Status Interoperable
+*/
+- (void)setCount:(NSUInteger)count {
+    if (count == _count) {
+        return;
+    }
+
+    if (count < _count) {
+        auto relinquishFunction = self.pointerFunctions.relinquishFunction;
+        if (relinquishFunction) {
+            for (NSUInteger i = count; i < _count; i++) {
+                if (_container[i]) {
+                    relinquishFunction(_container[i], sizeFunction);
+                    _container[i] = nullptr;
+                }
+            }
+        }
+    }
+
+    if (count > _allocated) {
+        _container = static_cast<void**>(realloc(_container, sizeof(void*) * count));
+        _allocated = count;
+    }
+
+    if (count > _count) {
+        memset(_container + _count, 0, sizeof(void*) * (count - _count));
+    }
+
+    _count = count;
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSArray*)allObjects {
+    NSMutableArray* array = [NSMutableArray arrayWithCapacity:_count];
+    for (NSUInteger i = 0; i < _count; i++) {
+        auto ptr = _container[i];
+        if (ptr != nullptr) {
+            [array addObject:static_cast<NSObject*>(ptr)];
+        }
+    }
+    return [NSArray arrayWithArray:array];
+}
+
+/**
+ @Status Interoperable
 */
 - (void*)pointerAtIndex:(NSUInteger)index {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (index < _count) {
+        return _container[index];
+    } else {
+        return nullptr;
+    }
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)addPointer:(void*)pointer {
-    UNIMPLEMENTED();
+    assert(_count <= _allocated && "Inconsistent internal state");
```

Comment:
I replaced with an exception, but I'm not sure about what you mean by checking on the reference platform: this is just a test to make sure that my implementation is not in a bad state, and I don't know how to reproduce such a state on OS X for instance.